src/mac_features/ack_generator/nrf_802154_ack_data.h

```diff
@@ -99,6 +101,18 @@ bool nrf_802154_ack_data_for_addr_clear(const uint8_t * p_addr, bool extended, u
  */
 void nrf_802154_ack_data_reset(bool extended, uint8_t data_type);
 
+/**
+ * @brief Select the source matching algorithm.
+ *
+ * @note This method should only be called in the initialization phase.
+ *
+ * When calling @ref nrf_802154_ack_data_pending_bit_should_be_set method, several different
+ * algorithms can be chosen to determine whether the pending bit should be set.
+ *
+ * @param[in]  match_method Source matching method to be used.
+ */
+void nrf_802154_ack_data_src_matching_method(nrf_802154_src_match_t match_method);
```

Comment:
The function name lacks a verb describing its purpose. It could inform explicitly that it's about the address. `nrf_802154_ack_data_src_addr_match_method_set` ?